pkg/network/namespace/namespace.go

```diff
@@ -146,11 +148,55 @@ func Delete(ns ns.NetNS) error {
 func Exists(name string) bool {
 	nsPath := filepath.Join(netNSPath, name)
 	_, err := os.Stat(nsPath)
-	return err == nil
+	if err != nil {
+		return false
+	}
+	mounted, err := isNamespaceMounted(name)
+	if err != nil {
+		log.Error().Err(err).Msg("failed to check if namespace is mounted")
+		return false
+	}
+
+	if !mounted {
+		//the file shouldn't be there
+		os.Remove(nsPath)
```

Comment:
make it explicit you don't want to handle the error
```suggestion
		_ = os.Remove(nsPath)
```